code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/api/action/SetServiceLinksActionHandler.java

```diff
@@ -0,0 +1,89 @@
+package io.cattle.platform.servicediscovery.api.action;
+
+import static io.cattle.platform.core.model.tables.ServiceConsumeMapTable.SERVICE_CONSUME_MAP;
+import io.cattle.platform.api.action.ActionHandler;
+import io.cattle.platform.core.model.Service;
+import io.cattle.platform.core.model.ServiceConsumeMap;
+import io.cattle.platform.json.JsonMapper;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.object.process.ObjectProcessManager;
+import io.cattle.platform.object.util.DataAccessor;
+import io.cattle.platform.servicediscovery.api.constants.ServiceDiscoveryConstants;
+import io.cattle.platform.servicediscovery.api.dao.ServiceConsumeMapDao;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class SetServiceLinksActionHandler implements ActionHandler {
+    
+    @Inject
+    JsonMapper jsonMapper;
+
+    @Inject
+    ObjectManager objectManager;
+
+    @Inject
+    ObjectProcessManager objectProcessManager;
+
+    @Inject
+    ServiceConsumeMapDao consumeMapDao;
+
+
+    @Override
+    public String getName() {
+        return ServiceDiscoveryConstants.PROCESS_SERVICE_SET_SERVICE_LINKS;
+    }
+
+    @Override
+    public Object perform(String name, Object obj, ApiRequest request) {
+        if (!(obj instanceof Service)) {
+            return null;
+        }
+        Service service = (Service) obj;
+        List<? extends Long> newConsumedServicesIds = DataAccessor.fromMap(request.getRequestObject())
+                .withKey(ServiceDiscoveryConstants.FIELD_SERVICE_IDS).asList(jsonMapper,
+                        Long.class);
+        if (newConsumedServicesIds != null) {
```

Comment:
I think it's odd that an empty list of ids and a null list of ids would behave differently. Calling this with a null serviceIds field would result in a no-op while calling it with an empty list would result in the services being removed.

Does it make sense to make the field not-nullable? Or did I miss something.